rholang/src/main/scala/coop/rchain/rholang/interpreter/Substitute.scala

```diff
@@ -13,98 +13,127 @@ import coop.rchain.models._
 import coop.rchain.models.rholang.sort._
 import coop.rchain.rholang.interpreter.errors.SubstituteError
 import coop.rchain.models.rholang.implicits._
-import errors._
-import coop.rchain.models.rholang.sort.ordering._
+import coop.rchain.models.rholang.sort._
+import coop.rchain.rholang.interpreter.accounting.CostAccountingAlg
```

Comment:
I'm uncertain about needing the cost accounting inside of substitution. We should be able to calculate it outside.

Reply:
Well, I don't like it because it pollutes the `Substitution` logic. If cost of substitution is:

1. Cost of the substituted `Par`  (if substitution is possible)
2. Cost of the initial `Par` (if no substitution)
3. Cost of the initial `Par` if an user error occurs

Then we would need to perform another equivalence check -- if what goes out from the `Substitute` equals what went it (and handle error cases). I will look at that possibility and see how it compares to the current one.

Reply:
Oh no, it's even simpler than that. It's always the _"cost"_ of what goes out of the substitution... Oh my, that's so obvious. Thanks @guardbotmk3 .